### DISCUSSION TITLE
Disable the predefined event handler

### DISCUSSION THREAD

**Discussion by ANtlord** [fltk-rs/fltk-rs#1399] [Q&A]
Hello,

I can't disable the default event handling for group::Scroll.  I need an area which I can scroll, but I want to prevent the default behavior of arrows buttons.  Is it possible somehow? I've checked the FLTK itself (I mean the C++ library itself, not cfltk) and looks like the only way to do it is making a child class out of FL_Scroll and redefine it's `int handle(int)` method.  In my case I just need it empty, but I can't get what I should do at Rust level.

**Comment by MoAlyousef** [comment]
Hello

The Fl_Scroll's handle method is nearly empty already. The arrow button behavior is actually from the Fl_Scrollbar widgets which are the first children of the Fl_Scroll. Since these are not created by fltk-rs, we actually have no way to safely override their handle method.

Another way to approach this is by stopping the propagation of the arrow keys from the Scroll to its scrollbar children:
```rust
use fltk::{prelude::*, enums::*, *};

fn main() {
    let a = app::App::default();
    let mut wind = window::Window::default().with_size(400, 300);
    let mut s = group::Scroll::default_fill();
    let mut b1 = button::Button::new(0, 0, 80, 30, "Hello1");
    let mut b2 = button::Button::new(0, 600, 80, 30, "Hello2");
    s.end();
    wind.end();
    wind.show();

    s.handle(|s, ev| match ev {
        Event::KeyDown => {
            let key = app::event_key();
            if key == Key::Down || key == Key::Up {
                // this stops the propagation of the event to the scroll's children
                true
            } else {
                false
            }
        },
        _ => false,
    });
    
    a.run().unwrap();
}
```

**Comment by ANtlord** [comment]
@MoAlyousef you are blazingly fast!  This doesn't work.  I've tried the entire snippet you've shared.  When I press arrows they change focus from the button `b1` to button `b2` and the other way around.  That's precisely the problem I have.  Is it possible to avoid it?

**Reply by MoAlyousef** [reply]
You can also pass app::set_visible_focus(false) and ignore the arrow keys at the level of the window itself:
```rust
use fltk::{prelude::*, enums::*, *};

fn main() {
    let a = app::App::default();
    app::set_visible_focus(false);
    let mut wind = window::Window::default().with_size(400, 300);
    let mut s = group::Scroll::default_fill();
    let mut b1 = button::Button::new(0, 0, 80, 30, "Hello1");
    let mut b2 = button::Button::new(0, 600, 80, 30, "Hello2");
    s.end();
    wind.end();
    wind.show();

    wind.handle(|_, ev| match ev {
        Event::KeyDown => {
            let key = app::event_key();
            if key == Key::Down || key == Key::Up {
                // this stops the propagation of the event to the scroll's children
                true
            } else {
                false
            }
        },
        _ => false,
    });
    
    a.run().unwrap();
}
```
